cha/implementaion/initial/ci.tex

```diff
@@ -1 +1,36 @@
-\section{Continous Integration}
+\section{Automated System Building with Continuous Integration}
+%We want to be assisted in integrating newly developed features etc.
+%Also, automate tests (dev env / target) and docs
+%Android APK availability
+%Hand in hand with agile
+While implementing our Android application, is it important that we do not get disconnected from the end product, albeit an unfinished one.
+This means that we need some way of continuously integrating new features, bug--fixes, and other changes into the code base, whilst also reflecting them in the product.
+However, we still want to have some assurance of our product working as intended, which means that we also need to perform some form of testing.
+
+\bigskip
+To meet these needs, we implement a so called \textit{continuous integration} or \textit{CI} into our development process.
+This CI includes conducting the following steps in a sandboxed environment, meaning that no state is saved across runs:
+\begin{enumberate}
+\item merging the new code into the existing code base, to check if it is possible;
+\item compiling the app and linking dependencies, such as external libraries;
+\item linting the newly added code, which entails checking the code style for bad practices and use of deprecated functionality in e.g. the Android APIs;
+\item performing tests, both unit testing but also device tests which requires running the application on an actual Android phone;
+\item generate documentation, based on the Javadoc\footnote{\url{https://en.wikipedia.org/wiki/Javadoc}} defined in comments throughout the code; and
+\item finally, packaging the Android app into a downloadable apk--file, for pre--release distribution.
+\end{enumberate}
+
+To carry out these tasks we use a piece of software called Jenkins\footnote{\url{https://jenkins.io/}}, which is a build automation server.
+The build and build process, our Android app in this case, is then defined in a \textit{Jenkinsfile}, which is a recipe with steps to follow for the Jenkins server.
+In our case, the output of the continuous integration using a Jenkins server, are test and linting reports, documentation, and the aforementioned packaged apk--file.
+
+The process of developing on our Android app, is based upon a workflow revolving around source code management using \textit{git}\footnote{\url{https://git-scm.com/}}, and project management using Phabricator\footnote{\url{https://www.phacility.com/}} and the Github\footnote{\url{https://github.com}} feature \textit{pull requests}.
+This development process will be further explained and discussed in \cnameref{cha:development_method}.
+
+We set up the continuous integration, such that any pull request and update to the master branch is activated the automated build system on a in--house Jenkins server.
```

Comment:
Er det ikke "is activated" -> "activates", eller misforstår jeg meningen med sætningen?